File: 2.py
import sys
def initList()->list:
    for index,line in enumerate(sys.stdin):
        if(index==1):
            return line.strip().split()

def fingMax(a_lt:list,minNum:int)->int:
    minList = 101
    sumNum = 0
    result = 0
    for i in a_lt:
        i = int(i)
        if(i<minNum):
            result = max(result,sumNum*minNum)
            sumNum = 0
            minList = 101
        else:
            minList = min(minList,i)
            sumNum += i
    return result

if __name__ == '__main__':
    a_lt = initList()
    # a_lt = [6,2,1]
    result = 0
    for i in range(100,1,-1):
        temp = fingMax(a_lt,i)
        result = max(temp,result)
    print("%d"%result)

File: 2.cpp
//
// Created by zb198 on 2019/8/23.
//
#include <vector>
#include <iostream>

void initVector(std::vector<int>& list,int n){
    for(int i=0;i<n;i++){
        scanf("%d",&list[i]);
    }
}

int FindMax(std::vector<int>& list,int minNum){
    int result=0;
    int sumNum = 0;
    int minList = 101;
    for(auto i:list){
        if(i>=minNum) {
            sumNum += i;
            minList = std::min(minList,i);
        } else {
            result = std::max(result,minList*sumNum);
            minList = 101;
            sumNum = 0;
        }
    }
    return result;
}

int main(){
    int n;
    scanf("%d",&n);
    std::vector<int> list(n);
    initVector(list,n);
    int maxNum = -1;
    for(int i=100;i>=1;i--){
        int temp = FindMax(list,i);
        if(temp > maxNum){
            maxNum = temp;
        }
    }
    printf("%d",maxNum);
}